backend/controller/internal/dal/dal.go

```diff
@@ -621,10 +612,33 @@ func (p *postgresClaim) Runner() Runner { return p.runner }
 
 // SetDeploymentReplicas activates the given deployment.
 func (d *DAL) SetDeploymentReplicas(ctx context.Context, key model.DeploymentName, minReplicas int) error {
```

Comment:
I think to make `tx.CommitOrRollback()` work correctly, you'll want to change the return signature to `(err error)`.